### PR TITLE
BRH-553

### PR DESCRIPTION
**Pull request by tianj7**
Jira Ticket: https://ctds-planx.atlassian.net/browse/BRH-553

### New Features
Add detailed workspace launch statuses

### CODE REVIEW

File: src/Workspace/index.jsx [hunk 1]
```diff
@@ -256,20 +256,50 @@ class Workspace extends React.Component {
       (element.type === 'ContainersReady' && element.status === 'False')
     ))) {
       workspaceLaunchStepsConfig.currentIndex = 2;
-      if (workspaceStatusData.containerStates.some((element) => (
+      workspaceLaunchStepsConfig.steps[2].description = 'In progress';
+      const cs = workspaceStatusData.containerStates;
+      console.log('container status length:' + cs.length);
```
Comment (mfshao) [inline src/Workspace/index.jsx:261]: seems to be unnecessary console output

File: src/Workspace/index.jsx [hunk 2]
```diff
@@ -256,20 +256,50 @@ class Workspace extends React.Component {
...
         (element.state && element.state.terminated)
       ))) {
         workspaceLaunchStepsConfig.steps[2].description = 'Error';
         workspaceLaunchStepsConfig.currentStepsStatus = 'error';
-      } else {
-        workspaceLaunchStepsConfig.steps[2].description = 'In progress';
+      }
+      else
+      {
+        if(cs.length > 1)
```
Comment (mfshao) [inline src/Workspace/index.jsx:271]: why not display these info if there is only 1 container?

File: src/Workspace/index.jsx (hunk 1 above)
Comment (tianj7) [reply src/Workspace/index.jsx]: thanks for catching this. Removed.

File: src/Workspace/index.jsx (hunk 2 above)
Comment (tianj7) [reply src/Workspace/index.jsx]: Changed the compare to > 0

File: src/Workspace/index.jsx [hunk 3]
```diff
@@ -256,20 +256,49 @@ class Workspace extends React.Component {
...
       return workspaceLaunchStepsConfig;
     }
 
     // here we are at step 3, step 3 have no k8s pod/container conditions
     workspaceLaunchStepsConfig.steps[0].description = 'Pod scheduled';
     workspaceLaunchStepsConfig.steps[1].description = 'Pod initialized';
+
+    // Display ECS status
+    if(workspaceStatusData.workspaceType == 'ECS')
+    {
```
Comment (nss10) [inline src/Workspace/index.jsx:289]: Use `===` and `!===` wherever possible

Comment (nss10) [reply src/Workspace/index.jsx:289]: replace `if(` with `if (` and prefer `if () {` instead of 
```
if
{
```

File: src/Workspace/index.jsx [hunk 4]
```diff
@@ -256,20 +256,49 @@ class Workspace extends React.Component {
...
+      {
+        //If container states are available, display detailed pod statuses
+        if(cs.length > 0)
+        {
+          for (let i = 0; i < cs.length; i++)
+          {
+            let j = i+1
+            workspaceLaunchStepsConfig.steps[2].description = workspaceLaunchStepsConfig.steps[2].description.concat(' \n Container ' + j + ' Ready: ' + cs[i].ready);
+          }
+        }
```
Comment (nss10) [inline src/Workspace/index.jsx:278]:
```suggestion
            } else if (cs.length > 0) {
        // If container states are available, display detailed pod statuses
        for (let i = 0; i < cs.length; i += 1) {
          const j = i + 1;
          workspaceLaunchStepsConfig.steps[2].description = workspaceLaunchStepsConfig.steps[2].description.concat(` \n Container ${j} Ready: ${cs[i].ready}`);
        }
      }
```

Comment (nss10) [reply src/Workspace/index.jsx:278]: Suggested some linting changes

**Review by nss10 — APPROVED** [review]
Logic looks good to me, some minor linting changes

**Review by nss10 — APPROVED** [review]

**Pull request merged**